packages/terra-table/src/subcomponents/_HeaderCheckMarkCell.jsx

```diff
@@ -112,6 +112,15 @@ const HeaderCheckMarkCell = ({
       attrSpread.onMouseDown = wrappedEventCallback(onMouseDown, event => event.currentTarget.setAttribute('data-cell-show-focus', 'false'));
     }
 
+    if (isSelected) {
```

Comment:
We should be able to reduce the amount of duplication here. aria-checked and aria-disabled may be able to be set outside of the selectable check.

Reply:
updated here 8d49a0f83d94b3b750db025fb58b2929599252ca